### PR TITLE
python: filter_by: use python filter and dont group array

### PR DESCRIPTION
**Pull request by brandsimon**
I think the python filter solution has a better performance and is more readable.

I dont know, if there is a language difference, but I think that the js function behaved differently than python/php. With python/php it was not possible to filter for values that evalutate to False, for example False, 0, ...

### CODE REVIEW

File: python/ccxt/base/exchange.py [hunk 1]
```diff
@@ -767,12 +767,7 @@ def deep_extend(*args):
 
     @staticmethod
     def filter_by(array, key, value=None):
-        if value:
-            grouped = Exchange.group_by(array, key)
-            if value in grouped:
-                return grouped[value]
-            return []
-        return array
+        return list(filter(lambda x: x[key] == value, array))
```
Comment (kroitor) [inline python/ccxt/base/exchange.py]: @brandsimon the issue with it is that it lost the ability to filter dicts now: https://github.com/ccxt/ccxt/issues/6150 ← this is obviously a problem with the `.orders` cache (which is `id`-indexed), the EXMO implementation (and a few other) will apply `filter_by` to the `.orders` cache...

Comment (kroitor) [reply python/ccxt/base/exchange.py]:
```
>>> list(filter(lambda x: x['a'] == 1, {'a':1, 'b': 2}))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <lambda>
TypeError: string indices must be integers
```

Comment (kroitor) [reply python/ccxt/base/exchange.py]: Will need to add `array = Exchange.to_array (array)` to python and php `filter_by`.

Comment (brandsimon) [reply python/ccxt/base/exchange.py]: @kroitor Yes, sorry, my bad. Thanks for notifying me and fixing, I hope I will remember it the next time I edit code like this.

Comment (kroitor) [reply python/ccxt/base/exchange.py]: @brandsimon it's not your fault, we just lack the tests for the python basecode... Will do our best to add them.

**Pull request merged**